### ISSUE TITLE
Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by legoboy0215** [iTXTech/Genisys#2096]
#### Issue description
This seg fault happens sometimes.

#### Steps to reproduce the issue

Start, and stop the server.

* Genisys: any, like literally any PM fork
* PHP: 7.0.12
* Server OS: Ubuntu
* Game version: Irrelevant

#### Crashdump, backtrace, memory dumps, plugins list or other files

Here is the coredump:
```
#0  0x00007f0d5e55ac37 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
#1  0x00007f0d5e55e028 in __GI_abort () at abort.c:89
#2  0x00007f0d5e5972a4 in __libc_message (do_abort=do_abort@entry=1, fmt=fmt@entry=0x7f0d5e6a56b0 "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:175
#3  0x00007f0d5e5a2b02 in malloc_printerr (ptr=0x2b8f3c0, str=0x7f0d5e6a17e4 "corrupted double-linked list", action=1) at malloc.c:4996
#4  malloc_consolidate (av=av@entry=0x7f0d5e8e2760 <main_arena>) at malloc.c:4157
#5  0x00007f0d5e5a345d in _int_free (av=0x7f0d5e8e2760 <main_arena>, p=<optimized out>, have_lock=0) at malloc.c:4057
#6  0x00000000007bc641 in zend_shutdown ()
#7  0x000000000074d76f in php_module_shutdown ()
#8  0x000000000043bd97 in main ()
```
Analyzed with gdb. No, I know seg faults have been around for a long time, but this happened suddenly.

**Comment by dktapps** [comment]
Does this happen on shutdown? After Stopping other threads... ?

**Comment by legoboy0215** [comment]
Yes. I can give a lot of different core dumps, I just need to do some tests.

**Comment by ishitatsuyuki** [comment]
pthreads upstream issues. *Don't care.*

**Comment by legoboy0215** [comment]
Well, it affects my restarts. It takes like 5 minutes for it to restart.

**Comment by ishitatsuyuki** [comment]
segfault is a instant crash and shouldn't take any time.

**Comment by legoboy0215** [comment]
No, I meam after that. When rhe seg fault happens, you know the -l opt in start.sh? the loop just takes a long time to start Genisys again.

**Comment by ishitatsuyuki** [comment]
It's not supposed to have that problem, check your configuration.

**Comment by dktapps** [comment]
Fixed one such segfault in b3ffa67e733db956de27478800e3082c035dc05f . If you still have this issue please let us know.
